test/Microsoft.AspNetCore.Rewrite.Tests/PatternSegments/GlobalRuleUrlSegmentTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Rewrite.Internal.PatternSegments;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Rewrite.Tests.PatternSegments
+{
+    public class GlobalRuleUrlSegmentTests
+    {
+        [Theory]
+        [InlineData("http", "localhost", 80, "", "http://localhost:80/")]
+        [InlineData("http", "localhost", 80, "/foo/bar", "http://localhost:80/foo/bar")]
```

Comment:
We need to check something that has escaped characters in the path, like %3A. Also add test for the query.